File: schemas/recipe-cooking/convex/ratings.ts
// ratings: User scores and optional text reviews for recipes.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const ratings = defineTable({
  recipeId: v.id("recipes"),
  userId: v.id("users"),
  score: v.number(),
  review: v.optional(v.string()),
  updatedAt: v.number(),
})
  .index("by_recipe_id_and_user_id", ["recipeId", "userId"])
  .index("by_user_id", ["userId"]);

File: schemas/recipe-cooking/spacetimedb/ratings.rs
// ratings: User scores and optional reviews for recipes.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

// Composite unique(recipe_id, user_id) — enforce in reducer logic

#[spacetimedb::table(name = ratings, public)]
pub struct Rating {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub recipe_id: String, // FK → recipes.id (cascade delete)

    #[index(btree)]
    pub user_id: String, // FK → users.id (cascade delete)

    pub score: i32,
    pub review: Option<String>,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
